### ISSUE TITLE
small 'current task' UI

### ISSUE THREAD

**Issue by hoylemd** [JonasJurczok/factorio-todo-list#47]
This is a bigger idea for down the road.

Allow users to turn on a 'current task' UI that shows them their top open assigned task at all times, separately from the list UI.  Since it'll only show a single task, it can be small.  All it needs is to display the task's text and possibly have a 'complete' button.

I'm not sure how useful this really is, since you can get the same result from opening the full list, but I think I'd like it, so suggestion :)

**Comment by Tarrke** [comment]
Come on let me type my words...

**Comment by Tarrke** [comment]
My comment on #48:
Using the todo list today I had to save and quit then load my game quite a lot of times. Sitting back at the computer I can't remember what to do then add to open the UI to check the first task. Maybe add a way to get a view on the first task assigned to me.

What do you think?

**Comment by hoylemd** [comment]
Yeah, that's pretty much exactly what I had in mind! I was thinking it could be a separate UI element from the button or list, maybe appearing on the right side of the screen or something.  But having it replace the button might be better.  @JonasJurczok What do you think?

**Comment by Tarrke** [comment]
I was also thinking about a new UI element in the upper right corner of the
screen.

Le mer. 8 août 2018 à 03:25, Michael D. Hoyle <notifications@github.com> a
écrit :

**Comment by JonasJurczok** [comment]
I love the idea.
I would also vote for the right corner. Unfortunately there is no way to
create draggable windows in factorio :(

Tarrke <notifications@github.com> schrieb am Mi., 8. Aug. 2018, 07:41:

**Comment by hoylemd** [comment]
I think it is possible, actually.  I'm fairly certain I was able to drag around a window while playing the other day, but my memory is very unreliable... I might be thinking about Kerbal Space Program...

**Comment by JonasJurczok** [comment]
At least a couple months ago it was not. And there was a thread in the
forum where a developer explained why this is not possible (different
render engines/levels of the UI). It sounded very final :D

So you can drag the in-game windows but not the mod created ones. Feel free
to try :)

Anyway a static window would be totally fine I think.

>

**Comment by hoylemd** [comment]
oh, That makes sense, I was probably thinking of a built-in dialog.  Darn. it would be nice if they made that API available.

**Comment by JonasJurczok** [comment]
Yep. But he said this will never happen because all mod code runs on the
server and suddenly Devs would be required to deal with the server/client
sync problems etc. And... They don't trust us to handle this correctly :D

As you can desync the whole game with this I understand their point...

>

**Comment by Tarrke** [comment]
Just try the mods :
* Hellmod
* FNEI

I'm sure Hellmod has a dragable in-game windows. Almost sure FNEI has a repositionable UI.

I'll begin work on this today I think.

**Comment by Tarrke** [comment]
Grrr still not good... I must work on my git skills... I'll try one more time, should be good this time.

**Comment by Tarrke** [comment]
Okay I should have something that is up to date and working quite nicely. The only problem is that the frame is on the left side of the button flow, not on the rigth side.
